### PR TITLE
basic setup + implement 'eessi init' + add GitHub Actions workflow

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
GitHub Actions workflow works fine (see https://github.com/boegel/eessi-cli/actions/runs/9430906635), but is not run yet here because it's not merged yet

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        echo "export PATH=$PREFIX/bin:$PATH" >> ~/env
+        echo "export PYTHONPATH=$(ls -d $PREFIX/lib/python*/site-packages)" >> ~/env
+
+    - name: Test init (without EESSI mounted)
+      run: |
+        source ~/env
+        eessi init 2>&1 | grep "^ERROR: /cvmfs/software.eessi.io/versions/2023.06 does not exist"
+
+    - name: Mount EESSI
+      uses: eessi/github-action-eessi@v3.1.0
```
Comment (bedroge) [inline .github/workflows/test.yml:33]:
```suggestion
      uses: eessi/github-action-eessi@e1f8f20638ea417a18d23ab29443ee34794ff900 # v3.1.0
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+permissions:
+  contents: read # to fetch code (actions/checkout)
+jobs:
+  ubuntu-minimal:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python: ['3.6', '3.7', '3.8', '3.9', '3.10', '3.11', '3.12']
+    steps:
+    - uses: actions/checkout@v4.1.4
```
[the remark this answers is not in the document]
Comment (bedroge) [reply .github/workflows/test.yml:13]:
```suggestion
    - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    - name: set up Python
+      uses: actions/setup-python@v4
```
Comment (bedroge) [inline .github/workflows/test.yml:15]:
```suggestion
      uses: actions/setup-python@b64ffcaf5b410884ad320a9cfac8866006a109aa # v4.8.0
```

Or (don't know if there was a specific reason to stick to v4?):
```suggestion
      uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
```

**Review by bedroge — CHANGES_REQUESTED** [review]

**Review by bedroge — APPROVED** [review]

**Pull request merged**